processing-tests/common/test/kotlin/com/livefront/sealedenum/compilation/visibility/ProtectedInterfaceSealedClass.kt

```diff
@@ -0,0 +1,164 @@
+package com.livefront.sealedenum.compilation.visibility
+
+import com.livefront.sealedenum.GenSealedEnum
+import org.intellij.lang.annotations.Language
+
+open class ProtectedInterfaceOuterClass {
+
+    protected interface ProtectedInterface
+
+    sealed class ProtectedInterfaceSealedClass :
+        com.livefront.sealedenum.compilation.visibility.JavaProtectedInterfaceSubclass(), ProtectedInterface {
+
+        object FirstObject : ProtectedInterfaceSealedClass()
+
+        object SecondObject : ProtectedInterfaceSealedClass()
+
+        @GenSealedEnum(generateEnum = true)
+        companion object
+    }
+}
+
+@Language("kotlin")
+val protectedInterfaceSealedClassGenerated = """
+package com.livefront.sealedenum.compilation.visibility
+
+import com.livefront.sealedenum.EnumForSealedEnumProvider
+import com.livefront.sealedenum.SealedEnum
+import com.livefront.sealedenum.SealedEnumWithEnumProvider
+import java.lang.Class
+import kotlin.Int
+import kotlin.String
+import kotlin.collections.List
+
+/**
+ * An isomorphic enum for the sealed class
+ * [ProtectedInterfaceOuterClass.ProtectedInterfaceSealedClass]
+ */
+public enum class ProtectedInterfaceOuterClass_ProtectedInterfaceSealedClassEnum(
+    sealedObject: ProtectedInterfaceOuterClass.ProtectedInterfaceSealedClass
+) : JavaProtectedInterfaceBaseClass.ProtectedInterface by sealedObject {
+    ProtectedInterfaceOuterClass_ProtectedInterfaceSealedClass_FirstObject(com.livefront.sealedenum.compilation.visibility.ProtectedInterfaceOuterClass.ProtectedInterfaceSealedClass.FirstObject),
```

Comment:
Do you need the whole `com.livefront.sealedenum.compilation.visibility.` here?

Reply:
I believe there was some case where not using the full, canonical name caused issues. Since it's generated code, nobody really should be looking at it too much anyway.